packages/flutter_tools/lib/src/test/web_test_compiler.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:package_config/package_config.dart';
+import 'package:process/process.dart';
+
+import '../artifacts.dart';
+import '../base/common.dart';
+import '../base/config.dart';
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/platform.dart';
+import '../build_info.dart';
+import '../bundle.dart';
+import '../compile.dart';
+import '../dart/language_version.dart';
+import '../web/bootstrap.dart';
+import '../web/memory_fs.dart';
+
+/// A web compiler for the test runner.
+class WebTestCompiler {
+  WebTestCompiler({
+    @required FileSystem fileSystem,
+    @required Logger logger,
+    @required Artifacts artifacts,
+    @required Platform platform,
+    @required ProcessManager processManager,
+    @required Config config,
+  }) : _logger = logger,
+       _fileSystem = fileSystem,
+       _artifacts = artifacts,
+       _platform = platform,
+       _processManager = processManager,
+       _config = config;
+
+  final Logger _logger;
+  final FileSystem _fileSystem;
+  final Artifacts _artifacts;
+  final Platform _platform;
+  final ProcessManager _processManager;
+  final Config _config;
+
+  Future<WebMemoryFS> initialize({
+    @required Directory projectDirectory,
+    @required String testOutputDir,
+    @required List<String> testFiles,
+    @required BuildInfo buildInfo,
+  }) async {
+    LanguageVersion languageVersion = LanguageVersion(2, 8);
```

Comment:
Should the initial value be a named constant somewhere, perhaps even shared with non-web?

Reply:
This is somewhat temporary until I can support opt in sound mode